File: processor/src/main/java/com/qualityminds/lazyval/processor/internal/codegen/beanvalidation/ResultHelper.java
package com.qualityminds.lazyval.processor.internal.codegen.beanvalidation;

import com.palantir.javapoet.JavaFile;
import com.qualityminds.lazyval.processor.spi.GeneratorResult;

import java.util.stream.Stream;

class ResultHelper {

    static Stream<GeneratorResult> toResultStream(JavaFile javaFile, String packageName, String className) {
        var metadata = new GeneratorResult.Metadata(packageName, className);
        var javaResult = new GeneratorResult.Java(metadata, javaFile.toString());
        var serviceLoaderResult = new GeneratorResult.ServiceLoader(
                new GeneratorResult.Metadata("jakarta.validation", "ConstraintValidator"),
                metadata
        );
        return Stream.of(javaResult, serviceLoaderResult);
    }
}

File: ksp/src/main/kotlin/com/qualityminds/lazyval/ksp/internal/codegen/beanvalidation/ResultHelper.kt
package com.qualityminds.lazyval.ksp.internal.codegen.beanvalidation

import com.qualityminds.lazyval.ksp.spi.GeneratorResult
import com.squareup.kotlinpoet.FileSpec
import java.util.stream.Stream

internal object ResultHelper {

    fun toResultStream(fileSpec: FileSpec, packageName: String, className: String): Stream<GeneratorResult> {
        val metadata = GeneratorResult.Metadata(packageName, className)
        val kotlinResult = GeneratorResult.Kotlin(metadata, fileSpec.toString())
        val serviceLoaderResult = GeneratorResult.ServiceLoader(
            GeneratorResult.Metadata("jakarta.validation", "ConstraintValidator"),
            metadata
        )
        return Stream.of(kotlinResult, serviceLoaderResult)
    }
}
